wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -161,10 +182,186 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
+                xs.map(out => processUtxo(out.index).map(_ -> out))
+              }
+
+          val createIncomingTxo: Seq[(SpendingInfoDb, OutputWithIndex)] => Vector[
+            IncomingWalletTXO] =
+            _.map {
+              case (utxo, OutputWithIndex(out, index)) =>
+                IncomingWalletTXO(
+                  confirmations = confirmations,
+                  // is this always the case?
+                  spent = false,
+                  scriptPubKey = out.scriptPubKey,
+                  txid = transaction.txIdBE,
+                  // always defined, as its freshly
+                  // written to the DB
+                  spendingInfoID = utxo.id.get
+                )
+            }.toVector
+
+          val incomingTXOsFut = addUTXOsFut.map(createIncomingTxo)
+          val writeIncomingTXOsFut =
+            incomingTXOsFut.flatMap(incomingTxoDAO.createAll)
+
+          writeIncomingTXOsFut
+
+      }
+    }
+  }
+
+  /**
+    * Processes an incoming transaction that already exists in our wallet.
+    * If the incoming transaction has more confirmations than what we
+    * have in the DB, we update the TX
+    */
+  private def processExistingIncomingTxo(
+      transaction: Transaction,
+      confirmations: Int,
+      foundTxo: IncomingWalletTXO): Future[Option[IncomingWalletTXO]] = {
+    if (foundTxo.confirmations < confirmations) {
+      logger.debug(
+        s"Increasing confirmation count of txo=${transaction.txIdBE}, old=${foundTxo.confirmations} new=${confirmations}")
+      val updateF =
+        incomingTxoDAO.update(foundTxo.copy(confirmations = confirmations))
+
+      updateF.foreach(tx =>
+        logger.debug(
+          s"Updated confirmation count=${tx.confirmations} of output=${foundTxo}"))
+      updateF.failed.foreach(err =>
+        logger.error(
+          s"Failed to update confirmation count of transaction=${transaction.txIdBE}",
+          err))
+
+      updateF.map(Some(_))
+    } else if (foundTxo.confirmations > confirmations) {
+      val msg =
+        List(
+          s"Incoming transaction=${transaction.txIdBE} has fewer confirmations=$confirmations",
+          s"than what we already have registered=${foundTxo.confirmations}! I don't know how",
+          s"to handle this."
+        ).mkString(" ")
+      logger.warn(msg)
+      Future.failed(new RuntimeException(msg))
+    } else {
+      logger.debug(
+        s"Skipping further processing of transaction=${transaction.txIdBE}, already processed.")
+      Future.successful(None)
+    }
+  }
+
+  override def processTransaction(
+      transaction: Transaction,
+      confirmations: Int): Future[LockedWallet] = {
+    logger.info(
+      s"Processing transaction=${transaction.txIdBE} with confirmations=$confirmations")
+
+    val incomingTxoFut: Future[Vector[IncomingWalletTXO]] =
+      incomingTxoDAO
+        .findTx(transaction)
+        .flatMap {
+          // no existing elements found
+          case Vector() =>
```

Comment:
`Vector.empty`

Reply:
`Vector.empty` is a `def` so it cannot be used in pattern-matching :-(